presto-native-execution/presto_cpp/main/PrestoServer.h

```diff
@@ -275,8 +275,16 @@ class PrestoServer {
   std::unique_ptr<PeriodicTaskManager> periodicTaskManager_;
   std::unique_ptr<PrestoServerOperations> prestoServerOperations_;
   std::unique_ptr<PeriodicMemoryChecker> memoryChecker_;
-  bool isMemOverloaded_{false};
-  bool isCpuOverloaded_{false};
+
+  // Last known memory overloaded status.
+  bool memOverloaded_{false};
+  // Last known CPU overloaded status.
+  bool cpuOverloaded_{false};
```

Comment:
Does cpu overloaded helps or not? Can we experiment without cpu overloaded in followup? Thanks!

Reply:
@xiaoxmeng 
We have two separate thresholds: one for memory and another for CPU.
Having either of them set to zero will disable it.
So it is possible to do both memory and CPU, only memory, only CPU and none.

Reply:
Also CPU monitoring was added because Deltoid use case was suffering comms issues due to high CPU usage.

I plan to upgrade the CPU overload with a period to protect from spiky CPU.
In other words we will have to have high CPU for N seconds before considered it overloaded.

But not in this change.